plugins/tech-radar/src/components/Radar/Radar.tsx

```diff
@@ -28,27 +28,53 @@ export type Props = {
   svgProps?: object;
 };
 
-const Radar = (props: Props): JSX.Element => {
-  const { width, height, quadrants, rings, entries } = props;
+const Radar = ({
+  width,
+  height,
+  quadrants,
+  rings,
+  entries,
+  ...props
+}: Props): JSX.Element => {
+  const [adjustedQuadrants, setAdjustedQuadrants] =
+    useState<Array<Quadrant>>(quadrants);
+  const [adjustedRings, setAdjustedRings] = useState<Array<Ring>>(rings);
+  const [adjustedEntries, setAdjustedEntries] = useState<Array<Entry>>(entries);
+
   const radius = Math.min(width, height) / 2;
 
   const [activeEntry, setActiveEntry] = useState<Entry>();
   const node = useRef<SVGSVGElement>(null);
 
-  // TODO(dflemstr): most of this can be heavily memoized if performance becomes a problem
-  adjustQuadrants(quadrants, radius, width, height);
-  adjustRings(rings, radius);
-  adjustEntries(entries, quadrants, rings, radius, activeEntry);
+  useEffect(() => {
+    setAdjustedQuadrants(adjustQuadrants(quadrants, radius, width, height));
+  }, [quadrants, radius, width, height]);
```

Comment:
I think all of these can be refactored into `useMemo` instead. The combination of a plain `useEffect` + `useState` is usually something you want to avoid. Probably want to merge them all into a single one as well